modules/apps/headless/headless-delivery/headless-delivery-test/src/testFunctional/macros/DocumentFolderAPI.macro

```diff
@@ -68,4 +177,67 @@ definition {
 		return ${response};
 	}
 
+	macro getDocumentFoldersWithDifferentPameter {
+		var portalURL = JSONCompany.getPortalURL();
+		var curl = DocumentFolderAPI._curlDocumentFolders(
+			groupName = ${groupName},
+			parameter = ${parameter},
+			parameterValue = ${parameterValue});
+
+		var response = JSONCurlUtil.get(${curl});
+
+		return ${response};
+	}
+
+	macro setupGlobalDocumentFolder {
+		Variables.assertDefined(parameterList = ${groupName});
+
+		if (isSet(externalReferenceCode)) {
+			var body = '''"name": "${name}","externalReferenceCode": "${externalReferenceCode}"''';
+		}
+		else {
+			var body = '''"name": "${name}"''';
+		}
+
+		var curlWithoutBody = DocumentFolderAPI._curlDocumentFolders(groupName = ${groupName});
+
+		var curl = StringUtil.add(${curlWithoutBody}, "-d {${body}}", " ");
+
+		var response = JSONCurlUtil.post(${curl});
+
+		var documentFolderId = JSONPathUtil.getIdValue(response = ${response});
+		var externalReferenceCode = JSONPathUtil.getErcValue(response = ${response});
+		static var staticResponse1 = ${response};
+		static var staticDoucmentFolderId = ${documentFolderId};
+		static var staticDoucmentFolderErc = ${externalReferenceCode};
+
+		return ${staticResponse1};
+
+		return ${staticDoucmentFolderId};
+
+		return ${staticDoucmentFolderErc};
+	}
+
+	macro updateDocumentFolderBySiteIdAndErc {
+		Variables.assertDefined(parameterList = ${groupName});
+
+		var portalURL = JSONCompany.getPortalURL();
+		var curlWithoutBody = DocumentFolderAPI._curlDocumentFolders(
+			externalReferenceCode = ${externalReferenceCode},
+			groupName = ${groupName});
+
+		if (isSet(name) && isSet(externalReferenceCodeInBody)) {
```

Comment:
I think externalReferneceCode in the body is not going to work, so we can define it "" in the boday if erc is not set:
`var body = '''"name": "${name}","externalReferenceCode": "${externalReferenceCodeInBody}"''';`